pkg/plugin/twinmaker/client.go

```
@@ -534,31 +534,26 @@ func (c *twinMakerClient) GetSessionToken(ctx context.Context, duration time.Dur
 }
 
 func (c *twinMakerClient) GetWriteSessionToken(ctx context.Context, duration time.Duration, workspaceId string) (*sts.Credentials, error) {
-	_, err := c.writerService()
-	if err != nil {
-		return nil, err
+	if c.tokenRoleWriter != "" {
```

failing early if the writer role is not configured